pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1586,19 +1586,17 @@ func containerAndPodFromLabels(inspect *dockertypes.ContainerJSON) (pod *api.Pod
 			if container == nil {
 				err = fmt.Errorf("unable to find container %s in pod %v", name, pod)
 			}
-		} else {
-			pod = nil
+			// As pod could't be nil, need not run the code below, we return directly here
```

Comment:
I'd rather not change this. Returning in a nested, long `if` block hinders the readability.